pkg/controllers/trusted_ca_bundle_controller.go

```diff
@@ -0,0 +1,213 @@
+package controllers
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/util"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	trustedCAConfigMapName      = "ccm-trusted-ca"
+	trustedCABundleConfigMapKey = "ca-bundle.crt"
+	systemTrustBundlePath       = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+)
+
+type TrustedCABundleReconciler struct {
+	client.Client
+	Scheme          *runtime.Scheme
+	Recorder        record.EventRecorder
+	TargetNamespace string
+	trustBundlePath string
+}
+
+// isSpecTrustedCASet returns true if spec.trustedCA of proxyConfig is set.
+func isSpecTrustedCASet(proxyConfig *configv1.ProxySpec) bool {
+	return len(proxyConfig.TrustedCA.Name) > 0
+}
+
+func (r *TrustedCABundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.Infof("%s emitted event, syncing %s ConfigMap", req, trustedCAConfigMapName)
+	var ccmTrustedConfigMap *corev1.ConfigMap
+
+	proxyConfig := &configv1.Proxy{}
+	if err := r.Get(ctx, types.NamespacedName{Name: proxyResourceName}, proxyConfig); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			klog.Infof("proxy not found; reconciliation will be skipped")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, fmt.Errorf("failed to get proxy '%s': %v", req.Name, err)
+	}
+
+	// check if changed config map in 'openshift-config' namespace is proxy trusted ca
+	if req.Namespace == OpenshiftConfigNamespace {
+		if proxyConfig.Spec.TrustedCA.Name != req.Name {
+			klog.Infof("changed config map %s is not a proxy trusted ca, skipping", req)
+			return reconcile.Result{}, nil
+		}
+	}
+
+	systemTrustBundle, err := r.getSystemTrustBundle()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get system trust bundle: %v", err)
+	}
+	ccmTrustedConfigMap = r.makeCABundleConfigMap(systemTrustBundle)
```

Comment:
Can remove the declared variable on line 48 I think
```suggestion
	ccmTrustedConfigMap := r.makeCABundleConfigMap(systemTrustBundle)
```